pkg/app/app.go

```diff
@@ -276,6 +276,38 @@ func (a *App) ConfigMapRefs() map[reftracker.RefKey]struct{} {
 	return configMaps
 }
 
+func (a *App) isSafeToFallbackToNoopDelete() bool {
+	if a.app.Status.Deploy == nil || a.app.Status.Deploy.KappDeployStatus == nil || a.app.Spec.Cluster != nil {
+		return false
+	}
+
+	// Ensure that app config map will be garbage collected due to namespace termination
+	status, err := a.compInfo.NamespaceStatus(a.app.Namespace)
```

Comment:
I think this namespace will be part of the list in the app status as kapp keeps the namespace where the cm is created as well. If we want to keep this to be on the safer side, I guess we can keep the `isNamespaceTerminating` method and pass the namespace as a parameter to avoid some duplicate code.

Reply:
> I think this namespace will be part of the list in the app status as kapp keeps the namespace
That does not seem to be the case, I just verified with a quick install. I think kapp only stores resources where it has to do list calls while grabbing resources. 

>  I guess we can keep the isNamespaceTerminating method and pass the namespace as a parameter to avoid some duplicate code.

Yep that sounds good!